xc/device.ml

```diff
@@ -2059,30 +1885,334 @@ module Dm = struct
     stop_vgpu ();
     stop_qemu ()
 
-  let is_hvm_linux domid =
-    let _XEN_IOPORT_LINUX_PRODNUM = 3 in (* from Linux include/xen/platform_pci.h *)
-    let error x = error "%s" x; Result.Error x in
-    try
-      let open Qmp in
-      match qmp_write_and_read domid (Command (None, Query_xen_platform_pv_driver_info)) with
-      | Some (Success (None, Xen_platform_pv_driver_info { product_num; build_num })) ->
-        Result.Ok ((product_num = _XEN_IOPORT_LINUX_PRODNUM) && (build_num <= 0xff))
-      | Some x ->
-        error (Printf.sprintf "Unexpected QMP response: %s" (Qmp.string_of_message x))
-      | None -> error "No QMP response for Query_xen_platform_pv_driver_info"
-    with e ->
-      error (Printf.sprintf "Exception attempting to obtain Linux product_id and build_number: %s" (Printexc.to_string e))
+end (* End of module Dm_Common *)
+
+
+(* Implementation of the qemu profile backends *)
+module Backend = struct
+
+  module type Intf = sig
+    module Vbd: sig
+      val qemu_media_change : xs:Xenstore.Xs.xsh -> device -> string -> string -> unit
+    end
+    module Dm: sig
+      module Event: sig
```

Comment:
Let's see if we can just remove this and leave the Qemu upstream Event mechanism as an implementation detail

Reply:
done